pkgs/by-name/my/mymake/package.nix

```
@@ -0,0 +1,43 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mymake";
+  version = "2.3.6";
+
+  src = fetchFromGitHub {
+    owner = "fstromback";
+    repo = "mymake";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-UQjvxdOvD9O6TrzBFJwh3CistDGZM9HZbcwVPx1n4+A=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    ./compile.sh mm
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp mm $out/bin
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Tool for compiling and running programs with minimal configuration.";
+    homepage = "https://github.com/fstromback/mymake";
+    maintainers = with maintainers; [ nobbele ];
+    license = licenses.mit;
+    platforms = platforms.linux;
```

minor nit but `with lib` increase eval time and is generally discouraged 
```suggestion
  meta = {
    description = "Tool for compiling and running programs with minimal configuration.";
    homepage = "https://github.com/fstromback/mymake";
    maintainers = with lib.maintainers; [ nobbele ];
    license = lib.licenses.mit;
    platforms = lib.platforms.linux;
```

---

if we are here removing `with`s I'm also a big fan of moving to
```
maintainers = [ lib.maintainers.nobbele ];
```

</very_minor_nit>

---

I also like not using `with` for single maintainers.

---

There's a lot of anti-patterns checked into nixpkgs just because of how old it is and how many different people do reviews. There's also just a lot of different opinions on how things should be done :D